Replace database dependency with expectations for tests

StevenMaude opened the pull request:
Fixes #93.

Use generated data from expectations framework if no database is available.

Previously there was an actual database containing synthetic data, but this has now been shutdown.

StevenMaude committed:
Switch from using SQL database

To expectations.

The database has been decommissioned.

StevenMaude committed:
Get CSV generation working

Add expectations to various variables.

StevenMaude commented:
These changes have been made primarily by referring to code snippets from other studies.

As far as I know, these are all MIT licensed (whether LSHTM or DataLab), so there should be no issue with this reuse.

This PR would also benefit from a review by a researcher.

StevenMaude committed:
Uncomment remaining lines

The workflow will fail with this change as we don't set expectations
yet.

StevenMaude commented:
The only thing blocking this is that now is I don't know what are suitable expectations [for these two entries](https://github.com/opensafely/risk-factors-research/blob/d58ef034e5b63fdde3bf7af4b724b0efdc43acae/analysis/study_definition.py#L337-L348). If they are commented out, then the tests pass. 

Also note that this isn't how things are done now. We could update to use the opensafely CLI, but it feels like:

* there's more danger of changing something so it isn't semantically as it was when this study was being actively developed
* it's preferable to capture the study as it was, if it is now complete

sebbacon commented:
Looking at the analytic code, neither `most_recent_unclear_smoking_numeric` nor `most_recent_unclear_smoking_cat` appear to be used; can we just delete them from the study definition?

StevenMaude committed:
Remove `most_recent_unclear_smoking_cat`/`numeric`

As noted by @sebbacon, these don't appear to be used in the study.

The lack of default expectations for these breaks the tests, and it is
not clear what those default expectations should be.

StevenMaude committed:
Retain option for generate cohort with database

Previously, the database option was just replaced with generated
expectations. It is unlikely that this option would be needed again, but
much else of the code and documentation refers to it, so leave it as is
for now.

StevenMaude commented:
As tidying, we can also delete the `DUMMY_DATABASE_URL` secret from this repository.

alexwalkerepi reviewed (approved):
All the expectations look fine to me, and so long as it's passing that's all we need. I even managed to get it to generate the dummy data locally.
